$OpenBSD: patch-CMakeLists_txt,v 1.3 2022/02/15 17:24:11 sthen Exp $

cmake: add Werror=unused-command-line-argument before testing cflags
c0dbc12df884fd9a76b4c6d6633dd48d41dc1517

Index: CMakeLists.txt
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -227,11 +227,19 @@ function(check_flag lang flag)
     string(REGEX REPLACE "[^A-Za-z0-9]" "_" flag_var "${flag}")
     if(NOT DEFINED ${lang}_FLAG${flag_var})
         execute_process(COMMAND ${CMAKE_COMMAND} -E echo_append "-- Checking ${lang} flag support for: [${flag}] - ")
+        if(NOT MSVC)
+            set(STORE_CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+            set(STORE_CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror=unused-command-line-argument")
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror=unused-command-line-argument")
+        endif()
         if("${lang}" STREQUAL "CXX")
             check_cxx_compiler_flag("${flag}" "${lang}_FLAG${flag_var}")
         else()
             check_c_compiler_flag("${flag}" "${lang}_FLAG${flag_var}")
         endif()
+        set(CMAKE_C_FLAGS "${STORE_CMAKE_C_FLAGS}")
+        set(CMAKE_CXX_FLAGS "${STORE_CMAKE_CXX_FLAGS}")
         if(${lang}_FLAG${flag_var})
             execute_process(COMMAND ${CMAKE_COMMAND} -E echo "Yes")
         else()
